### PR TITLE
[Cache] remove Doctrine DBAL < 3.3 related code

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

following the merge up of #52459

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/DoctrineDbalAdapter.php [hunk 1]
```diff
@@ -375,8 +375,7 @@ private function getServerVersion(): string
             return $this->serverVersion;
         }
 
-        // The condition should be removed once support for DBAL <3.3 is dropped
-        $conn = method_exists($this->conn, 'getNativeConnection') ? $this->conn->getNativeConnection() : $this->conn->getWrappedConnection();
+        $conn = $this->conn->getNativeConnection();
         if ($conn instanceof ServerInfoAwareConnection) {
```
Comment (derrabus) [inline src/Symfony/Component/Cache/Adapter/DoctrineDbalAdapter.php:379]: Hmm, this does not really make sense. If we call `getNativeConnection()`, we get the native handle of the PHP extension used, e.g. a `mysqli` instance or an `sqlsrv` resource. Those objects/resources won't ever implement `ServerInfoAwareConnection`. Something's off here.

Comment (xabbuh) [reply src/Symfony/Component/Cache/Adapter/DoctrineDbalAdapter.php:379]: I guess you are right. Looks like we first need to fix it in a way similar to #52715.

**Comment by derrabus** [comment]
My alternative proposal: #52720.

**Comment by xabbuh** [comment]
closing in favour of #52720

**Pull request closed without merging**